### ISSUE TITLE
Outdated lodash-es

### ISSUE THREAD

**Issue by FritzHerbers** [lodash/lodash#4879]
>Looking for Lodash modules written in ES6 or smaller bundle sizes? Check out lodash-es.

Is there any reason why the NPM package versions for lodash and lodash-es differ ?

lodash: 4.17.19
lodash-es: 4.17.15

**Comment by alllx** [comment]
I also noticed that there are no latest changes in the NPM:
```
dist
.tarball: https://registry.npmjs.org/lodash/-/lodash-4.17.19.tgz
.shasum: e48ddedbe30b3321783c5b4301fbd353bc1e4a4b
```
 like this one: https://github.com/lodash/lodash/pull/4623/files which being merged in to master.
Could you please update the NPM package please.

**Comment by jrrbru** [comment]
We also rely on this for an Ember app and would appreciate it being updated to the latest version. Let me know if there's any way I can help. We'd be happy to investigate build issues or something if it meant we could stay up to date.

**Comment by sgomes** [comment]
As an extra data point, we rely on there being matching versions of `lodash` and `lodash-es` to perform some [`webpack` module replacements](https://github.com/Automattic/wp-calypso/tree/master/packages/webpack-extensive-lodash-replacement-plugin) in [wp-calypso](https://github.com/Automattic/wp-calypso), with the goal of replacing all `lodash` usage (and `lodash.foo` usage) with `lodash-es`. This helps us avoid shipping both packages, since we have runtime dependencies that themselves depend on either one of them.

As it is, we'll likely start having issues once a dependency decides to make use of `lodash` > 4.17.15, so an updated version of `lodash-es` could save us a lot of trouble.

Happy to help in any way I can, as well!

**Comment by zmazouzi** [comment]
I am having the same issue where I cannot upgrade `lodash-es` to 4.17.20 due to https://nvd.nist.gov/vuln/detail/CVE-2020-8203
and using lodash would lead to bigger bundle files. An updated version of `lodash-es` could save us a lot of trouble. 
Thank you.

**Comment by dlangerenken** [comment]
Can the priority for this ticket please be bumped? It's impossible to use lodash-es with those vulnerabilities. This is a blocker for our project.

**Comment by sgomes** [comment]
@mathiasbynens Sorry to summon you to this thread, but I noticed you have deploy rights on the npm package. Would you happen to know how we could get a new version of `lodash-es` released, to match the current `lodash`? The security vulnerabilities are becoming a concern 😕

**Comment by mathiasbynens** [comment]
I don't have any more information than you all do. I don't know how to cut a `lodash-es` release.

**Comment by sgomes** [comment]
Thank you for getting back to us, @mathiasbynens 👍 Sorry again for pulling you in like that; it was a bit of a long shot, but I figured it was worth a try.

**Comment by dlangerenken** [comment]
@jdalton do you happen to know how to cut a release for lodash-es? You're the one tagging the last lodash-es version.

**Comment by noinkling** [comment]
Note that at https://lodash.com/docs the "latest" version in the top-left is 4.17.15 also, seems like another symptom of the release process not happening properly.

**Comment by rmedraoui** [comment]
We rely on lodash-es for our project. It will be very appreciated if you could update it to the latest version.

**Comment by mathiasbynens** [comment]
@bnjmnt4n Just invited you to the lodash npm org. Please let me know once you've accepted it!

**Comment by bnjmnt4n** [comment]
@mathiasbynens I just accepted the invite!

**Comment by SarathPravallika** [comment]
> I believe I can publish a new version of lodash-es, but I don't currently have publish rights on npm. /cc @mathiasbynens if you grant me the rights, I should be able to do so ~this week.

Any update on the publishing of latest version of lodash-es.... Our project security scans found vulnerabilities in 4.17.15 version, it would be much appreciated if the latest version can be pushed at the earliest possible. Thanks in advance

**Comment by scott-r-lindsey** [comment]
@bnjmnt4n Are you blocked on releasing to NPM?

**Comment by opichals** [comment]
This looks like it is causing quite a bit of a headache to `lodash` users.

The issues is that the latest version of [`lodash-es` npm module](https://www.npmjs.com/package/lodash-es) which has more than a [couple of thousands of dependents](https://www.npmjs.com/package/lodash-es) has a fixed dependency on a [vulnerability containing](https://snyk.io/vuln/npm:lodash) `lodash@4.17.15` version.

@jdalton @mathiasbynens As you are listed as contributors on npmjs.org. Is there anything I can help with to get the lodash & lodash-es bundles in sync in the public npm registry?

**Comment by mathiasbynens** [comment]
`GOTO` https://github.com/lodash/lodash/issues/4879#issuecomment-698971728

**Comment by mathiasbynens** [comment]
`GOTO` https://github.com/lodash/lodash/issues/4879#issuecomment-698971728 -- the main problem is it's unclear how to properly build and publish `lodash-es` (in a way that matches the previous releases). If someone figures it out, we can follow those steps and publish the update.

**Comment by opichals** [comment]
I seem to have been able to create something that resembles what is published in npm.

from node_modules/lodash-es/README.md:
```md
# lodash-es v4.17.15

The [Lodash](https://lodash.com/) library exported as [ES](http://www.ecma-international.org/ecma-262/6.0/) modules.

Generated using [lodash-cli](https://www.npmjs.com/package/lodash-cli):
```shell
$ lodash modularize exports=es -o ./
```

See the [package source](https://github.com/lodash/lodash/tree/4.17.15-es) for more details.
```

So I did:
```
git clone git@github.com:lodash-archive/lodash.git
git clone git@github.com:lodash-archive/lodash-cli.git
cd lodash-cli
npm install
node ./bin/lodash modularize exports="es" -o ./lodash-es`
```

Looks the same as in my other project node_modules/lodash folder. IMO Needs adding the README, LICENSE & package.json files.

**Comment by opichals** [comment]
@mathiasbynens Created a PR for the `es` branch with the generated result https://github.com/lodash/lodash/pull/4972

**Comment by nerdkid93** [comment]
@opichals is correct, when running `git diff --no-index` between `lodash-es` and the lodash `es` branch, the only differences are things that `npm publish` will ignore (e.g. `.git/`, `.gitignore`, `.npmignore`). The only thing that needs to be done to resolve this issue IMO is to merge #4947 into the `es` branch, then someone with `npm` credentials just needs to run `npm publish` from the updated `es` branch, and the new version will be published.

**Comment by opichals** [comment]
and https://github.com/lodash-archive/lodash-cli/pull/143

**Comment by nerdkid93** [comment]
> the main problem is it's unclear how to properly build and publish lodash-es (in a way that matches the previous releases)

@mathiasbynens these steps have already been completed in #4947. If you check out that base branch [lodash-es-cve-2020-8203](https://github.com/nerdkid93/lodash/tree/lodash-es-cve-2020-8203), and `diff` it with the published `tar` in NPM,  you'll see that the only differences are the changes made in lodash releases following the `v4.17.15` release.

**Comment by scott-r-lindsey** [comment]
It seems like we've got some movement here?  Is there anything we can do to help?

**Comment by breezeling** [comment]
Currently we are using lodash-es in our project. Version 4.17.15 has vulnerabilities which is a big headache .... really appreciate that if the new version can be published. :)

**Comment by matiassantos** [comment]
any updates on this issue?

**Comment by opichals** [comment]
One mitigation strategy till this is resolved could be to alias `lodash-es` to `lodash` in your `package.json`. While it doesn't solve bundle deduplication (the use of `lodash-es` doesn't either) it can help with mitigating the possible vulnerability impact.

```json
     "lodash": "=4.17.20",
     "lodash-es": "npm:lodash@4.17.20",
```

**Comment by sgomes** [comment]
@opichals : That mitigation strategy will definitely work, but it comes at the cost of bundle size. By using the unmodularised `lodash` you'll be including the entire library in your bundle, whereas with `lodash-es` you might be shipping only a small portion, depending on how much of the library you're using — even though the codebase is excessively interdependent, with simple methods like `_.once` pulling in a good chunk of the library.

`lodash` and `lodash-es` appear to be increasingly unsupported, so it seems the best long-term strategy is to move away from these libraries. ES2015+ and a polyfill strategy should be enough to provide a significant chunk of `lodash` functionality with a small enough level of development overhead, for anyone able to do a large-scale migration.

**Comment by bnjmnt4n** [comment]
Update: I've published [v4.17.20](https://github.com/lodash/lodash/releases/tag/4.17.20-es) of lodash-es to npm. Apologies for the wait, do create an issue if you encounter any issue. I'll also be posting an update with detailed instructions on how to perform the build steps for publishing new versions of lodash and lodash-es. (There were some updates to lodash-cli which prevented me from simply merging the changes in the other PRs). Thanks everyone for their assistance!
